### ISSUE TITLE
Modify/Sanitize filename

### ISSUE THREAD

**Issue by tmwangi** [waynehoover/s3_direct_upload#161]
Is there a way to modify the basename/filename of the s3 key? I realize that one can specify the key option in the s3_uploader_form function but what if i wanted to modify the {filename}. I would probably want to use to sanitize filenames(remove/replace spaces and special characters) instead of rejecting the users upload on the basis of invalid filename.
If there is no feature, can this functionality be added?

Thanks.

**Comment by ilyakatz** [comment]
:+1:

**Comment by ilyakatz** [comment]
Actually, looks like the latest version allows for cleaned_filename https://github.com/waynehoover/s3_direct_upload/blob/master/README.md#options-for-form-helper

But ... latest version on rubygems is almost a year and half ago 
![image](https://cloud.githubusercontent.com/assets/282514/10580280/c398a1dc-764a-11e5-9b95-29a8eaa5c9dd.png)

**Comment by tmwangi** [comment]
No longer working on a project that uses this. But if I get sometime i will
test it.

On 19 October 2015 at 07:18, Ilya Katz <notifications@github.com> wrote:

> Actually, looks like the latest version allows for cleaned_filename
> https://github.com/waynehoover/s3_direct_upload/blob/master/README.md#options-for-form-helper
>
> But ... latest version on rubygems is almost a year and half ago
> [image: image]
> <https://cloud.githubusercontent.com/assets/282514/10580280/c398a1dc-764a-11e5-9b95-29a8eaa5c9dd.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/waynehoover/s3_direct_upload/issues/161#issuecomment-149226367>
> .
>



-- 
tim

**Comment by trostli** [comment]
yes the new version from github is working for me to clean the filename
